File: go/tree/sumNumbers.go
package tree

func sumNumbers(root *TreeNode) int {
	var dfs func(node *TreeNode, path int) int
	dfs = func(node *TreeNode, path int) int {
		if node == nil {
			return 0
		}
		path = path*10 + node.Val
		if node.Left == nil && node.Right == nil {
			return path
		}
		return dfs(node.Left, path) + dfs(node.Right, path)
	}
	return dfs(root, 0)
}

File: java/src/main/java/io/dure/tree/SumNumbers.java
package io.dure.tree;

public class SumNumbers {
    public int sumNumbers(TreeNode root) {
        return dfs(root, 0);
    }

    private int dfs(TreeNode node, int path) {
        if (node == null) {
            return 0;
        }
        path = path * 10 + node.val;
        if (node.left == null && node.right == null) {
            return path;
        }
        return dfs(node.left, path) + dfs(node.right, path);
    }
}
